include/mbgl/style/conversion.hpp

```diff
@@ -1,306 +1,22 @@
-       excessive code bloat and long compile times.
-     * Runtime polymorphism using virtual methods requires extra heap allocation and ubiquitous
-       use of std::unique_ptr, unsuitable for this performance-sensitive code.
-
-   Therefore, we're using a custom implementation of runtime polymorphism where we manually create and
-   dispatch through a table of function pointers (vtable), while keeping the storage for any of the possible
-   underlying types inline on the stack, using `std::aligned_storage`.
-
-   For a given underlying type T, an explicit specialization of `ConversionTraits<T>` must be provided. This
-   specialization must provide the following static methods:
-
-      * `isUndefined(v)` -- returns a boolean indication whether `v` is undefined or a JSON null
-
-      * `isArray(v)` -- returns a boolean indicating whether `v` represents a JSON array
-      * `arrayLength(v)` -- called only if `isArray(v)`; returns a size_t length
-      * `arrayMember(v)` -- called only if `isArray(v)`; returns `V` or `V&`
-
-      * `isObject(v)` -- returns a boolean indicating whether `v` represents a JSON object
-      * `objectMember(v, name)` -- called only if `isObject(v)`; `name` is `const char *`; return value:
-         * is true when evaluated in a boolean context iff the named member exists
-         * is convertable to a `V` or `V&` when dereferenced
-      * `eachMember(v, [] (const std::string&, const V&) -> optional<Error> {...})` -- called
-         only if `isObject(v)`; calls the provided lambda once for each key and value of the object;
-         short-circuits if any call returns an `Error`
-
-      * `toBool(v)` -- returns `optional<bool>`, absence indicating `v` is not a JSON boolean
-      * `toNumber(v)` -- returns `optional<float>`, absence indicating `v` is not a JSON number
-      * `toDouble(v)` -- returns `optional<double>`, absence indicating `v` is not a JSON number
-      * `toString(v)` -- returns `optional<std::string>`, absence indicating `v` is not a JSON string
-      * `toValue(v)` -- returns `optional<Value>`, a variant type, for generic conversion,
-        absence indicating `v` is not a boolean, number, or string. Numbers should be converted to
-        unsigned integer, signed integer, or floating point, in descending preference.
-
-   In addition, the type T must be move-constructable. And finally, `Convertible::Storage`, a typedef for
-   `std::aligned_storage_t`, must be large enough to satisfy the memory requirements for any of the
-   possible underlying types. (A static assert will fail if this is not the case.)
-
-   `Convertible` itself is movable, but not copyable. A moved-from `Convertible` is in an invalid state;
-   you must not do anything with it except let it go out of scope.
-*/
-
 struct Error { std::string message; };
 
 template <typename T>
 class ConversionTraits;
 
-class Convertible {
```

Comment:
Add a comment here explaining that this is a minimal header suitable for obtaining forward declarations, and that implementation files that need to actually perform conversions should include `conversion_impl.hpp`.